### DISCUSSION TITLE
Support for receiving traces over Forward Input Plugin

### DISCUSSION THREAD

**Discussion by epsteina16** [fluent/fluent-bit#5870] [Ideas]
With upcoming support for traces in Fluent Bit 2.0, it would be good to be able to send OTEL traces to Fluent Bit over the Forward protocol. The mspacked data would need to follow the OTEL distributed traces data model. This would allow for existing applications that use msgpack to use Fluent Bit as an OTEL traces router.

**Comment by epsteina16** [comment]
@agup006 

Following up on our discussion yesterday during the community meeting. I propose, and am open to any and all comments, the following changes to the forward input plugin:
- A new parameter, called "data_type" that takes one of "logs" or "traces".
- When data_type is set to "logs", there are no changes to the Forward input plugin.
- When data_type is set to "traces", the forward input plugin will attempt to read OTLP JSON packed as msgpack over the configured input. Forward input will parse the msgpack data and inspect it to ensure it (a) is OTLP JSON data and (b) has the required fields in the OTLP spec. If the data received is invalid, for any reason, the Forward input plugin will still send an ACK back (since the data was received) but will not append the chunk to the engine and will instead log a data format error.
- Multiple forward input plugin instances can be used over different TCP/unix domain sockets, but this proposal does not allow for using one forward input plugin for both logs and traces.

**Comment by agup006** [comment]
@edsiper @leonardo-albertovich @pwhelan I recall some discussion on this topic though it would be good to update @epsteina16

**Comment by leonardo-albertovich** [comment]
I think this should be addressed in the log record improvement design document and it'd be great if we could do that before 2.0 of course.
